### ISSUE TITLE
Segmentation fault: Builder killed by signal SIGSEGV[11]

### ISSUE THREAD

**Issue by lilianmoraru** [tailhook/vagga#116]
Vagga version: v0.4.0-54-g331f7f0

The entire config file:
```
containers:

  ##[Helpers] Don't use ubuntu container directly, it's only for SubConfigs
  ubuntu:
    setup:
    - !UbuntuRelease { version: 15.10 }

  ##[Helpers] Don't use qt5 container directly, it's only for SubConfigs
  qt5:
    setup:
    - !SubConfig
      container: ubuntu
    - !Install [qt5-default, libgl1-mesa-dev]

  dev-env:
    setup:
    - !SubConfig
      container: qt5

commands:
  build: !Command
    container: dev-env
    description: Build QMLChecker with Qt 5.4.2
    run: "qmake --version"

  shell: !Command
    container: dev-env
    description: Open bash inside the container
    run: /bin/bash
```

Error:
```
ERROR:vagga::wrapper: Error executing _build: Builder killed by signal SIGSEGV[11] (core dumped)
Command <Command "/proc/self/exe" ("vagga_wrapper") "_build" "dev-env"; environ: {"ftp_proxy"="[company ftp proxy]","RUST_LOG"="warn","http_proxy"="http://localhost:3128","TERM"="xterm","VAGGA_USER_HOME"="/home/[my company user]","https_proxy"="https://localhost:3128",}; uid_map=[UidMap { inside_uid: 0, outside_uid: 1000, count: 1 }, UidMap { inside_uid: 100000, outside_uid: 100000, count: 65536 }]; gid_map=[GidMap { inside_gid: 0, outside_gid: 1000, count: 1 }, GidMap { inside_gid: 100000, outside_gid: 100000, count: 65536 }]> exited with code 124
```

Moving `!Install` in `ubuntu` container and changing in `dev-env` to use container `ubuntu` solves the issue. I wanted to separate the concerns...

What's even more strange, after it installs everything into the container I can change the config file as it was before and it will work ok. If I remove `.vagga` and run it again, it crashes again.
Segmentation fault even though using Rust :)
Searched now and you do have `unsafe` in a few locations. For some reason, I searched a while back and had the impression you had no `unsafe` inside the code(libraries using unsafe, not vagga itself).

Said `core dumped` but I couldn't manage to find it.

**Comment by tailhook** [comment]
It's probably stack overflow. Unfortunately musl libc has small stack (80kib) but rust usually relies on large stack (order of megabytes). I could not find easy way to change the stack size previously.

Anyway, if you can reproduce it reliably, I'll fix it.

In the meantime, you can probably run `vagga _build ubuntu && vagga _build qt5 && vagga _build dev-env`. I.e. build containers sequentially.

**Comment by lilianmoraru** [comment]
To reproduce it reliably you have to use this config file and do
`rm -rf .vagga && vagga build`

**Comment by tailhook** [comment]
Hm, I don't believe I didn't think of it before. With 80k stack and 64k default buffer for file copying (on stack) it surprising that it was working before :)

Fixed in v0.4.1-4-gf63d8a5
